src/custom-elements/src/components/icon/icon.tsx

```diff
@@ -15,7 +15,7 @@ export class Icon {
   /**
    * OPTIONAL. Color of the icon. Accepts any color the text-utility supports {primary, success, warning, danger, light, muted, grey, secondary}
    */
-  @Prop({ reflect: true }) color: string;
+  @Prop({ reflect: true }) color?: string;
```

Comment:
I have checked `icon.tsx` in more detail and I could see validateColorAttribute function wouldn't validate a `null` value. You'll need to update that as well. 
The other option would be to set a default color for being used when the developer doesn't specify a color himself like this:
```Javascript
@Prop({ reflect: true }) color = "something";
```
This might not work with the `color`, that the developer might want it to take the color from the context it is located, so the other approach is preferred. But, it might be the way to go for `size` and `extraClass`:
```Javascript
  @Prop({ reflect: true }) size = "md";
  @Prop({ reflect: true }) extraClass = "";
```